dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -1149,6 +1158,10 @@ SegmentPtr Segment::merge(DMContext & dm_context, const ColumnDefinesPtr & schem
     return merged;
 }
 
+/// Segments may contain some rows that not belong to its range which is left by previous split operation.
```

Comment:
Do you think split needs flushCache?

Reply:
We only need to flush segments either after split or before merge. And considering segment split mostly means it's a hot segment and segment merge means it's a cold segment. So flush before merge is a better choice which should do less flush work in most cases.